packages/replicache/src/replicache-impl.ts

```diff
@@ -1449,13 +1449,19 @@ export class ReplicacheImpl<MD extends MutatorDefs = {}> {
   #register<Return extends ReadonlyJSONValue | void, Args extends JSONValue>(
     name: string,
     mutatorImpl: (tx: WriteTransaction, args?: Args) => MaybePromise<Return>,
-  ): (args?: Args) => Promise<Return> {
+  ): (
+    args?: Args,
+  ) => Promise<Return> | {client: Promise<Return>; server: Promise<unknown>} {
```

Comment:
I think this should only return:

```
{client: Promise<Return>; server: Promise<Return>}
```

Is there a reason I'm not understanding to keep the `Promise<Return> |`  return option?

Reply:
I suppose we can do whatever we want with this return type because Replicache forcibly casts it later anyway:

https://github.com/rocicorp/mono/blob/8b97014070fc8333ffdfa548e7caee29674b921d/packages/replicache/src/replicache-impl.ts#L1491-L1493

and `Zero` does a similar thing by casting `rep.mutate` to `any` 

https://github.com/rocicorp/mono/blob/8b97014070fc8333ffdfa548e7caee29674b921d/packages/zero-client/src/client/zero.ts#L621

then assigning to the public interface.

https://github.com/rocicorp/mono/blob/8b97014070fc8333ffdfa548e7caee29674b921d/packages/zero-client/src/client/zero.ts#L646

---

`Promise<Return> | {client: Promise<Return>; server: Promise<unknown>}` would let us tighten these unsafe casts.

Reply:
The unsafe casts are there so the Replicache interface can remain unchanged.